backend/src/main/java/com/github/JeTSkY1h/user/MyUserController.java

```diff
@@ -31,10 +31,11 @@ ResponseEntity<Object> giveAdmin(@RequestBody MyUser newAdmin){
 
     @GetMapping("/bookdata")
     ResponseEntity<List<BookUserData>>getBookUserData(Principal principal){
-        if(myUserService.getBookuserData(principal.getName()).isEmpty()) {
+        Optional<MyUser> user = myUserService.findByUsername(principal.getName());
+        if(user.isEmpty()) {
             return null;
         }
-        return ResponseEntity.of(myUserService.getBookuserData(principal.getName()));
+        return ResponseEntity.ok(user.get().getBookData());
```

Comment:
Why did you change that from `ResponseEntity.of` to `ResponseEntity.ok`? I would remove the `return null` and rewrite the methode like this:
```java
return ResponseEntity.of(myUserService.findByUsername(principal.getName())
     .map(user -> user.getBookData()));
```
That way you'll receive a 200 with the book data, if the user is present and a 404 if not.